tests/pytest/sparse_ci/regex/test_regex.py

```diff
@@ -12,7 +12,7 @@ def test_regex():
     gop.add("[1a+ 0a-]", 1.0)
     test_str = gop.str()
     print(test_str)
-    ref_str = "+ [ 1a+ 0a- ]"
+    ref_str = "(1 + 0i) * [1a+ 0a-]"
```

Comment:
For real numbers, this practice of writing everything as complex does seem verbose... What do you think of changing the string representation to omit the real part for imaginary numbers and the imaginary part for real numbers?

Reply:
I agree, but I have to think more about printing. Deciding how to format numbers nicely is a complicated subject since there are many exceptions to consider. It will have to be part of a future PR.